mmtrack/datasets/pipelines/transforms.py

```diff
@@ -134,13 +136,267 @@ def __call__(self, results):
+                normalize=False)
+
+            generated_bbox = generated_bbox[None]
+
+            _results['img'] = crop_img
+            if 'img_shape' in _results:
+                _results['img_shape'] = crop_img.shape
+            _results[_results.get('bbox_fields', [])[0]] = generated_bbox
+            _results['seg_fields'] = ['att_mask']
+            _results[_results.get('seg_fields', [])[0]] = att_mask
+            outs.append(_results)
+        return outs
+
+
+@PIPELINES.register_module()
+class SeqBboxJitter(object):
+    """Bounding box jitter augmentation. The jittered bboxes are used for
+    subsequent image cropping, like `SeqCropLikeStark`.
+
+    Args:
+        scale_jitter_factor (list[int | float]): contains the factor of scale
+            jitter.
+        center_jitter_factor (list[int | float]): contains the factor of center
+            jitter.
+        crop_size_factor (list[int | float]): contains the ratio of crop size
+            to bbox size.
+    """
+
+    def __init__(self, scale_jitter_factor, center_jitter_factor,
+                 crop_size_factor):
+        self.scale_jitter_factor = scale_jitter_factor
+        self.center_jitter_factor = center_jitter_factor
+        self.crop_size_factor = crop_size_factor
+
+    def __call__(self, results):
+        """Call function.
+
+        Args:
+            results (list[dict]): list of dict from
+                :obj:`mmtrack.base_sot_dataset`.
+
+        Returns:
+            list[dict]: list of dict that contains augmented images.
+        """
+        outs = []
+        for i, _results in enumerate(results):
+            gt_bbox = _results[_results.get('bbox_fields', [])[0]][0]
+            x1, y1, x2, y2 = np.split(gt_bbox, 4, axis=-1)
+            bbox_w, bbox_h = x2 - x1, y2 - y1
+            gt_bbox_cxcywh = np.concatenate(
+                [x1 + bbox_w / 2., y1 + bbox_h / 2., bbox_w, bbox_h], axis=-1)
+
+            crop_img_size = -1
+            # avoid croped image size too small.
+            count = 0
+            while crop_img_size < 1:
+                count += 1
+                if count > 100:
+                    print(f'-------- bbox {gt_bbox_cxcywh} is invalid -------')
+                    return None
+                jittered_wh = gt_bbox_cxcywh[2:4] * np.exp(
+                    np.random.randn(2) * self.scale_jitter_factor[i])
+                crop_img_size = np.ceil(
+                    np.sqrt(jittered_wh.prod()) * self.crop_size_factor[i])
+
+            max_offset = np.sqrt(
+                jittered_wh.prod()) * self.center_jitter_factor[i]
+            jittered_center = gt_bbox_cxcywh[0:2] + max_offset * (
+                np.random.rand(2) - 0.5)
+
+            jittered_bboxes = np.concatenate(
+                (jittered_center - 0.5 * jittered_wh,
+                 jittered_center + 0.5 * jittered_wh),
+                axis=-1)
+
+            _results['jittered_bboxes'] = jittered_bboxes[None]
```

Comment:
Do we need we record both `jittered_bboxes` and `gt_bboxes`?
If not, just use `_results['gt_bboxes'] = jittered_bboxes[None]`

Reply:
Yes, we need record `both jittered_bboxes` and `gt_bboxes`.
`jittered_bboxes` is used to crop patches from the original image in `SeqCropLikeStark`, while `gt_bboxes` is used to record the true boxes.

Reply:
Do you need to use `gt_bboxes` after `SeqBboxJitter` class? If not, recording `gt_bboxes` is unnecessary.

Reply:
We need

Reply:
OK